src/hooks/useKeyboardNavigation.ts

```diff
@@ -0,0 +1,313 @@
+      itemsPerRow,
+      numOfItems,
+      currentColumn,
+      gridChildren
+    } = getGridInfo();
+
+    const {
+      isTopRow,
+      isBottomRow,
+      isLastRow,
+      isLeftColumn,
+      isRightColumn
+    } = getElementBoundariesInfo({ activeIndex, itemsPerRow, numOfItems });
+
+    switch (direction) {
+      case UP:
+        if (
+          isTopRow &&
+          !focusPrevEmojiListGroup(currentColumn, itemsPerRow ?? -1)
+        )
+          focusPrevSection();
+        else updateActiveItem(gridChildren?.[activeIndex - itemsPerRow]);
+        break;
+      case DOWN:
+        if (isBottomRow) {
+          if (isLastRow) focusNextEmojiListGroup(currentColumn);
+          else updateActiveItem(gridChildren?.[numOfItems - 1]);
+        } else updateActiveItem(gridChildren?.[activeIndex + itemsPerRow]);
+        break;
+      case LEFT: {
+        const prevEmoji = getPrevEmoji();
+
+        if (isLeftColumn)
+          prevEmoji
+            ? focusElement(prevEmoji)
+            : focusPrevEmojiListGroupOnLastItem();
+        else focusElement(prevEmoji);
+
+        break;
+      }
+      case RIGHT: {
+        const nextEmoji = getNextEmoji();
+
+        if (isRightColumn)
+          nextEmoji ? focusElement(nextEmoji) : focusNextEmojiListGroup();
+        else focusElement(nextEmoji);
+
+        break;
+      }
+    }
+  });
+
+  let sections: any[] = [];
+
+  useEffect(() => {
+    sections = [
+      {
+        //categories
+        focus: () => focusElement(categoriesNavRef.current.firstChild),
+        rootElement: categoriesNavRef
+      },
+      emojiSearchRef.current && {
+        // search bar
+        focus: () => focusElement(emojiSearchRef.current),
+        rootElement: emojiSearchRef
+      },
+      {
+        //emoji list'
+        focus: () => {
+          const firstEmoji = emojiListRef.current.querySelector('.epr-emoji');
+          focusElement(firstEmoji);
+        },
+        rootElement: emojiListRef
+      }
+    ].filter(Boolean);
+  }, []);
+
+  const focusNextEmojiListGroup = (columnIndex = 0) => {
+    const currentEmojiGroup = getCurrentEmojiListGroup();
+    const nextEmojiGroup = currentEmojiGroup?.nextSibling as HTMLElement;
+
+    if (nextEmojiGroup) {
+      focusElement(nextEmojiGroup.children[columnIndex]);
+
+      const categoryName = getEmojiGroupName(nextEmojiGroup);
+      setCategory(categoryName);
+    }
+  };
+
+  const focusPrevEmojiListGroupOnLastItem = () => {
+    const currentEmojiGroup = getCurrentEmojiListGroup();
+    const prevEmojiGroup = currentEmojiGroup?.previousSibling;
+    focusElement(prevEmojiGroup?.lastChild);
+    return prevEmojiGroup;
+  };
+
+  const focusPrevEmojiListGroup = (
+    columnIndex: number,
+    itemsPerRow: number
+  ) => {
+    const currentEmojiGroup = getCurrentEmojiListGroup();
+    const prevEmojiGroup = currentEmojiGroup?.previousSibling as HTMLElement;
+
+    if (prevEmojiGroup) {
+      const numOfItems = prevEmojiGroup.children.length;
```

Comment:
Note that there will always be children, since the emoji category label is an element as well. We need to think how to exclude it from the searches.